tsl/test/sql/compression_sequence_num_removal.sql

```diff
@@ -0,0 +1,154 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+-- test query planning with hypertable which contains
+-- compressed chunks that depend on sequence number optimization
+-- which is removed in latest schema revision
+\set EXPLAIN 'EXPLAIN (VERBOSE, COSTS OFF)'
+
+CREATE TABLE hyper(
+    time INT NOT NULL,
+    device_id INT,
+    val INT);
+SELECT * FROM create_hypertable('hyper', 'time', chunk_time_interval => 10);
+
+-- test case with segmentby
+ALTER TABLE hyper SET (
+    timescaledb.compress,
+    timescaledb.compress_orderby = 'time',
+    timescaledb.compress_segmentby = 'device_id');
+INSERT INTO hyper VALUES (1, 1, 1), (2, 2, 1), (3, 3, 1), (10, 3, 2), (11, 4, 2), (11, 5, 2), (21, 2, 3), (22, 3, 3), (23, 4, 3), (30, 1, 4), (31, 3, 4), (31, 5, 4);
+
+SELECT compress_chunk(show_chunks('hyper'));
+:EXPLAIN SELECT * FROM hyper
+ORDER BY device_id, time;
+
+-- modify two chunks by adding sequence number to the segments
+-- and rebuild the index based on that column
+SELECT comp_ch.table_name AS "CHUNK_NAME", comp_ch.schema_name|| '.' || comp_ch.table_name AS "CHUNK_FULL_NAME"
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.chunk comp_ch, _timescaledb_catalog.hypertable ht
+WHERE ch1.hypertable_id = ht.id AND ht.table_name LIKE 'hyper'
+AND ch1.compressed_chunk_id = comp_ch.id
+ORDER BY ch1.id LIMIT 1 \gset
+SELECT schemaname || '.' || indexname AS "CHUNK_INDEX" FROM pg_indexes where tablename = :'CHUNK_NAME'
+LIMIT 1 \gset
+
+\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER
+SET timescaledb.restoring TO ON;
+-- add sequence number column and fill in the correct sequences
+ALTER TABLE :CHUNK_FULL_NAME ADD COLUMN _ts_meta_sequence_num int;
+INSERT INTO :CHUNK_FULL_NAME (device_id, time, val, _ts_meta_count, _ts_meta_min_1, _ts_meta_max_1, _ts_meta_sequence_num)
+SELECT device_id, time, val, _ts_meta_count, _ts_meta_min_1, _ts_meta_max_1 , 10*row_number() over (partition by device_id) as _ts_meta_sequence_num
+FROM :CHUNK_FULL_NAME
+ORDER BY device_id, _ts_meta_min_1, _ts_meta_max_1;
+DELETE FROM :CHUNK_FULL_NAME WHERE _ts_meta_sequence_num IS NULL;
+-- drop exising index
+DROP INDEX :CHUNK_INDEX;
+-- create index based on sequence numbers
+CREATE INDEX ON :CHUNK_FULL_NAME (device_id, _ts_meta_sequence_num);
+SET timescaledb.restoring TO OFF;
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
```

Comment:
Unless you really want to start a new session, I recommend just using `SET ROLE` and `RESET ROLE` from here on. Once you are superuser, you can switch back and forth between privileged and unprivileged roles.

The reason for this is  that it becomes easier to debug the test if a crash would occur because you can run the sql on a standalone instance and not have the script kill your sessions.